src/semver.ts

```diff
@@ -27,17 +41,21 @@ export function determineSemverChange(
   return hasMajor ? "major" : hasMinor ? "minor" : hasPatch ? "patch" : null;
 }
 
+export type BumpVersionOptions = {
+  type?: SemverBumpType;
+  preid?: string;
+};
+
 export async function bumpVersion(
   commits: GitCommit[],
   config: ChangelogConfig,
-  opts: { type?: SemverBumpType } = {}
+  opts: BumpVersionOptions = {}
 ): Promise<string | false> {
   let type = opts.type || determineSemverChange(commits, config) || "patch";
   const originalType = type;
 
-  const pkgPath = resolve(config.cwd, "package.json");
-  const pkg =
-    JSON.parse(await fsp.readFile(pkgPath, "utf8").catch(() => "{}")) || {};
+  const pkgPath = await resolvePackageJSON(config.cwd);
```

Comment:
I think we should keep  fully resolved `pkgPath` as absolute path passing to `resolvePackageJSON` and `writePackageJSON` (otherwise we might end-up updating an unwanted file)

Reply:
You mean we should set pkgPath in context (or config)?

Reply:
Basically use resolve rather than resolvePackageJSON same as beforr

Reply:
OK ! get that